docs/upgrading-openshift.md

```diff
@@ -0,0 +1,100 @@
+
+{: .no_toc }
+
+1. TOC
+{:toc}
+
+## From `rhcos` Version 0.1.0 to `openshift` Version 4.8.0
+
+The new `openshift` config variant is intended to work both on the OpenShift Container Platform with RHEL CoreOS, and on OKD with Fedora CoreOS. The `rhcos` variant is still accepted by Butane but will not receive new features.
+
+The `openshift` 4.8.0 specification is not backward-compatible with the `rhcos` 0.1.0 specification. It adds new mandatory metadata fields and removes certain Ignition config fields. In addition, `openshift` configs are transpiled to an OpenShift [MachineConfig] rather than an Ignition config by default. A valid `rhcos` 0.1.0 configuration can be updated to an `openshift` 4.8.0 configuration by changing the variant and version strings and then correcting any errors reported during transpilation.
+
+The following is a list of breaking changes and notable new features.
+
+### MachineConfig generation
+
+By default, Butane transpiles an `openshift` Butane config into an OpenShift [MachineConfig]. Butane produces an Ignition config if the `-r` or `--raw` option is specified on the Butane command line.
+
+### Removed config fields
+
+The config no longer allows certain Ignition config fields that are rejected or discouraged by the OpenShift [Machine Config Operator].
+
+In the `storage` section, `directories` and `links` are removed, along with `append` in `files`. Local file trees referenced in `trees` must not contain symlinks.
+
+In the `passwd` section, `groups` is removed. All fields in `users` are removed except for `name` (which must be set to `core`) and `ssh_authorized_keys`.
+
+### MachineConfig metadata fields
+
+The config gained a new top-level `metadata` section containing metadata for the generated [MachineConfig]. The mandatory `name` field specifies a [name for the Kubernetes MachineConfig resource][k8s-names]. The `labels` field specifies a map of key-value pairs to be applied to the MachineConfig resource as [Kubernetes labels][k8s-labels]. The `machineconfiguration.openshift.io/role` label is required.
+
+The `metadata` section is ignored when generating a raw Ignition config using the `-r` or `--raw` option.
+
+<!-- butane-config -->
+```yaml
+variant: openshift
+version: 4.8.0
+metadata:
+  name: minimal-config
+  labels:
+    machineconfiguration.openshift.io/role: worker
+```
+
+### MCO settings
+
+The config gained a new top-level `openshift` section specifying [configuration][MCO settings] for the [Machine Config Operator]. The `extensions` field lists RHCOS extension modules to be installed on the node. The `fips` field enables [FIPS 140-2] mode when set to `true`. The `kernel_arguments` field specifies a list of arguments to be added to the kernel command line. The `kernel_type` field can be set to `realtime` to use the realtime kernel variant on the node.
```

Comment:
May be useful to link to the extensions docs here - https://docs.openshift.com/container-platform/4.7/post_installation_configuration/machine-configuration-tasks.html#rhcos-add-extensions_post-install-machine-configuration-tasks

Reply:
Argh, I see the link to that overall set of docs now...

I guess I shouldn't do CR before my first coffee :grimacing:

Reply:
It's a good point, though.  Added some additional links.